### PR TITLE
Ruby on Rails Project: Forms

### PR DESCRIPTION
**Pull request by jmontanholi**
# In this project I created the following:
  * ### Users Model and Controller.
  * ### Users/new view.
  * ### Users/:id/edit view.
  * ### new, edit, update, create and user_params methods.
  * ### Parameters validations.

### CODE REVIEW

File: app/views/users/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<%= form_with model:@user do |form| %>
```
Comment (Dipeshtwis) [inline app/views/users/new.html.erb:36]: [Optional]

- [x] form view to displays a list of the error messages (with #errors and full_messages usage) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_extra_shared_error.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms)).

File: app/views/users/edit.html.erb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+<% if @user.errors.any? %>
+  <div id="error_explanation">
```
Comment (Dipeshtwis) [inline app/views/users/edit.html.erb:2]: Good job here. You have implemented the feature that shows the error message.

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
# Status:  ♻️ Changes Required ♻️
 Hi @jmontanholi

**You've really hit all the requirements in one shot. Well done.** Here're the major highlights of this PR:
- Your PR title and description are really descriptive.
- The project is working as expected.

**However, in order to approve, we have to work a little bit more. Please check my comments below and fix them as appropriate.**

## For Approval

- [x] In Readme, please write all the steps one needs to follow to run in the localhost. [refer this](https://github.com/Dipeshtwis/Bare_Metal_Forms_and_Helpers/tree/feature/feature_form). For now, I had to figure out myself that I have to do `bundle install`, `rails db:migrate`, in order to run the project.
- [x] There should be commented code for **form** ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form1.png) [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form2.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms)) and **form_tag** ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form_tag2.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms)), so, that we can see how you've applied.

## [Optional]

- [ ] partial created for user form and reused in edit and new view
- [ ] validate the email format by using either a regular expression or a built-in URI method.

Remember, This is optional and nothing to do with the project approval.

## Please make the required changes and submit another review request when you are done.

(Feel free to contact me on slack by the handle of  **deep** if you have any additional questions).
Happy Coding
[Dipesh Kumar](https://deeptwist.tk) | [Twitter](https://twitter.com/97deepeshkumar)

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
# Status:  ♻️ Changes Required ♻️
 Hi @jmontanholi 

Please fix the above changes requested before submitting another code review. [☝️ check here](https://github.com/jmontanholi/rails_project_forms/pull/1#pullrequestreview-649535742)


## Please make the required changes and submit another review request when you are done.

(Feel free to contact me on slack by the handle of  **deep** if you have any additional questions).
Happy Coding
[Dipesh Kumar](https://deeptwist.tk) | [Twitter](https://twitter.com/97deepeshkumar)

**Comment by jmontanholi** [comment]
Sorry, I forgot to push the changes.

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
### Status: ♻️ Changes Required ♻️
**I'm happy to see the changes requested being implement, but two things have prevented this project from being approved.**
- [x] Study about how to comment in Ruby.
- [ ] In the readme file, explain how to clone a project, just like you are explaining to a person who does not know about how to clone a project.

**Review by DeliceLydia — APPROVED** [review]
# Status: Approved image

Congratulations @jmontanholi on implementing all the changes requested by the previous TSE's your project met requirements and am super honored to approve it

## Feel free to merge and move on to the next project

Regards,

Lydia Ingabire
Slack: Delice Lydia
[Linkedin](https://www.linkedin.com/in/delice-lydia/)

**Pull request merged**